test/Core.Test/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommandTests.cs

```diff
@@ -0,0 +1,631 @@
+            .GetByIdAsync(organizationId)
+            .Returns(organization);
+
+        var organizationUpdate = new OrganizationUpdate
+        {
+            OrganizationId = organizationId,
+            MaxAutoscaleSeats = 15,
+            SeatAdjustment = 5,
+            MaxAutoscaleServiceAccounts = 300,
+            ServiceAccountsAdjustment = 100
+        };
+        var plan = StaticStore.SecretManagerPlans.FirstOrDefault(x => x.Type == organization.PlanType);
+        plan.HasAdditionalSeatsOption = true;
+        plan.HasAdditionalServiceAccountOption = true;
+        var additionalSeats = (organization.SmSeats + organizationUpdate.SeatAdjustment) - plan.BaseSeats;
+        var additionalServiceAccounts =
+            (organization.SmServiceAccounts + organizationUpdate.ServiceAccountsAdjustment) - plan.BaseServiceAccount;
+        var newSeatTotal = organization.SmSeats.HasValue
+            ? organization.SmSeats.Value + organizationUpdate.SeatAdjustment
+            : organizationUpdate.SeatAdjustment;
+
+        var newServiceAccountsTotal = organization.SmServiceAccounts.HasValue
+            ? organization.SmServiceAccounts.Value + organizationUpdate.ServiceAccountsAdjustment
+            : organizationUpdate.ServiceAccountsAdjustment;
+
+        await sutProvider.Sut.UpdateSecretsManagerSubscription(organizationUpdate);
+
+        await sutProvider.GetDependency<IPaymentService>().Received(1)
+            .AdjustSeatsAsync(organization, plan, additionalSeats.GetValueOrDefault());
+
+        await sutProvider.GetDependency<IPaymentService>().Received(1)
+            .AdjustServiceAccountsAsync(organization, plan, additionalServiceAccounts.GetValueOrDefault());
+
+        await sutProvider.GetDependency<IReferenceEventService>().Received(1).RaiseEventAsync(
+            Arg.Is<ReferenceEvent>(referenceEvent =>
+                referenceEvent.Type == ReferenceEventType.AdjustSmSeats &&
+                referenceEvent.Id == organization.Id &&
+                referenceEvent.PlanName == plan.Name &&
+                referenceEvent.PlanType == plan.Type &&
+                referenceEvent.Seats == newSeatTotal &&
+                referenceEvent.PreviousSeats == previousSeats
+            )
+        );
+
+        await sutProvider.GetDependency<IReferenceEventService>().Received(1).RaiseEventAsync(
+            Arg.Is<ReferenceEvent>(referenceEvent =>
+                referenceEvent.Type == ReferenceEventType.AdjustServiceAccounts &&
+                referenceEvent.Id == organization.Id &&
+                referenceEvent.PlanName == plan.Name &&
+                referenceEvent.PlanType == plan.Type &&
+                referenceEvent.ServiceAccounts == newServiceAccountsTotal &&
+                referenceEvent.PreviousServiceAccounts == previousServiceAccounts
+            )
+        );
+
+        await sutProvider.GetDependency<IOrganizationService>().Received(1).ReplaceAndUpdateCacheAsync(organization);
```

Comment:
@cyprain-okeke / @r-tome correct me if I'm wrong, but I think this only verifies using the object reference. i.e. it checks that you passed the same `organization` object. It doesn't check its properties. I recommend using `Arg.Is` to make sure that we're inspecting the org's properties (`SmSeats` etc) and making sure that they're set correctly.

Reply:
@eliykat you're completely right!

Reply:
resolved